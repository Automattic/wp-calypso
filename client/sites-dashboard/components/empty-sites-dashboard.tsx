import { useI18n } from '@wordpress/react-i18n';
import EmptyContent from 'calypso/components/empty-content';
import { MEDIA_QUERIES } from '../utils';
import { CreateSiteCTA, MigrateSiteCTA } from './sites-dashboard-ctas';

const EmptySitesCTAs = () => {
	const { __ } = useI18n();

	return (
		<EmptyContent
			css={ {
				display: 'flex',
				flexDirection: 'column',
				alignItems: 'center',
				margin: 0,
				padding: 0,
				[ MEDIA_QUERIES.small ]: {
					width: '100%',
					maxWidth: '100%',
					alignItems: 'flex-start',
				},
			} }
			title={
				<span
					css={ {
						color: 'var(--studio-gray-100)',
						fontSize: '3rem',
						fontWeight: 400,
						[ MEDIA_QUERIES.small ]: { fontSize: '2rem', textAlign: 'left' },
						fontFamily: 'Recoleta, sans-serif',
					} }
				>
					{ __( 'Letâ€™s add your first site' ) }
				</span>
			}
			illustration=""
		>
			<div
				css={ {
					width: '85%',
					display: 'flex',
					gap: '32px',
					flexDirection: 'column',
					[ MEDIA_QUERIES.small ]: {
						width: '100%',
						gap: '24px',
					},
				} }
			>
				<CreateSiteCTA />
				<div
					css={ {
						'&:before': {
							content: '""',
							display: 'block',
							height: '1px',
							opacity: 0.64,
							background: '#DCDCDE',
						},
					} }
				/>
				<MigrateSiteCTA />
			</div>
		</EmptyContent>
	);
};

export const EmptySitesDashboard = () => {
	return (
		<div
			css={ {
				paddingTop: '20vh',
				[ MEDIA_QUERIES.small ]: {
					paddingTop: '60px',
				},
				zIndex: 9999,
				position: 'relative',
				display: 'flex',
				justifyContent: 'center',
			} }
		>
			<EmptySitesCTAs />
		</div>
	);
};
