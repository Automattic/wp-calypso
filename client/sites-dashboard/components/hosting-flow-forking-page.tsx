import { useI18n } from '@wordpress/react-i18n';
import EmptyContent from 'calypso/components/empty-content';
import { MEDIA_QUERIES } from '../utils';
import { CreateSiteCTA, MigrateSiteCTA } from './hosting-flow-forking-ctas';

interface HostingFlowForkingPageProps {
	siteCount: number;
}

export const HostingFlowForkingPage = ( { siteCount }: HostingFlowForkingPageProps ) => {
	const { __ } = useI18n();

	return (
		<EmptyContent
			css={ {
				display: 'flex',
				flexDirection: 'column',
				alignItems: 'center',
				margin: 0,
				padding: 0,
				gap: '32px',
				[ MEDIA_QUERIES.small ]: {
					width: '100%',
					maxWidth: '100%',
					alignItems: 'flex-start',
				},
			} }
			title={
				<span
					css={ {
						color: 'var(--studio-gray-100)',
						fontSize: '3rem',
						fontWeight: 400,
						[ MEDIA_QUERIES.small ]: { fontSize: '2rem', textAlign: 'left' },
						fontFamily: 'Recoleta, sans-serif',
					} }
				>
					{ siteCount === 0 ? __( 'Let’s add your first site' ) : __( 'Let’s add a site' ) }
				</span>
			}
			illustration=""
		>
			<div
				css={ {
					display: 'flex',
					flexDirection: 'column',
					[ MEDIA_QUERIES.small ]: {
						width: '100%',
					},
				} }
			>
				<CreateSiteCTA />
				<div
					css={ {
						margin: '32px 0',
						[ MEDIA_QUERIES.small ]: {
							margin: '24px 0',
						},

						'&:before': {
							content: '""',
							display: 'block',
							height: '1px',
							opacity: 0.64,
							background: '#DCDCDE',
						},
					} }
				/>
				<MigrateSiteCTA />
			</div>
		</EmptyContent>
	);
};
