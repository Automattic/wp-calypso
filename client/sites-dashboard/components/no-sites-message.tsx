import { localizeUrl } from '@automattic/i18n-utils';
import { createInterpolateElement } from '@wordpress/element';
import { useI18n } from '@wordpress/react-i18n';
import NoSitesMessageLayout from 'calypso/components/empty-content/no-sites-message';

type SitesContainerProps = {
	status: string;

	// The number of sites the user has that match the specified status
	statusSiteCount: number;
};

export const NoSitesMessage = ( { status, statusSiteCount }: SitesContainerProps ) => {
	const { __ } = useI18n();

	if ( statusSiteCount > 0 ) {
		// If the user has some number of sites with this status, but the table is
		// still empty, it means the search query must not match any sites.
		return <h2>{ __( 'No sites match your search.' ) }</h2>;
	}

	if ( status === 'public' ) {
		return (
			<NoSitesMessageLayout
				title={ __( "You haven't launched a site" ) }
				line={ createInterpolateElement(
					__(
						'Our <a>support center</a> and team are here to help you as you work your way towards launch.'
					),
					{
						a: (
							<a
								href={ localizeUrl( 'https://wordpress.com/support/' ) }
								target="_blank"
								rel="noopener noreferrer"
							/>
						),
					}
				) }
				illustration={ false }
			/>
		);
	}

	if ( status === 'private' ) {
		return (
			<NoSitesMessageLayout
				title={ __( 'You have no private sites' ) }
				line={ createInterpolateElement(
					__( "Private sites aren't accessible to the world. Read more about them <a>here</a>." ),
					{
						a: (
							<a
								href={ localizeUrl( 'https://wordpress.com/support/settings/privacy-settings/' ) }
								target="_blank"
								rel="noopener noreferrer"
							/>
						),
					}
				) }
				illustration={ false }
			/>
		);
	}

	if ( status === 'coming-soon' ) {
		return (
			<NoSitesMessageLayout
				title={ __( 'You have no coming soon sites' ) }
				line={ createInterpolateElement(
					__(
						'Coming soon sites will display a landing page letting people know that a site is being built. Read more about them <a>here</a>.'
					),
					{
						a: (
							<a
								href={ localizeUrl( 'https://wordpress.com/support/settings/privacy-settings/' ) }
								target="_blank"
								rel="noopener noreferrer"
							/>
						),
					}
				) }
				illustration={ false }
			/>
		);
	}

	if ( status === 'redirect' ) {
		return (
			<NoSitesMessageLayout
				title={ __( 'You have no redirected sites' ) }
				line={ createInterpolateElement(
					__(
						'Redirected sites send a visitor directly to the mapped domain. Read more about them <a>here</a>.'
					),
					{
						a: (
							<a
								href={ localizeUrl( 'https://wordpress.com/support/site-redirect/' ) }
								target="_blank"
								rel="noopener noreferrer"
							/>
						),
					}
				) }
				illustration={ false }
			/>
		);
	}

	return (
		<NoSitesMessageLayout
			title={ __( 'Create your first site' ) }
			action={ __( 'Create your first site' ) }
			actionURL="/start?source=sites-dashboard&ref=calypso-nosites"
		/>
	);
};
