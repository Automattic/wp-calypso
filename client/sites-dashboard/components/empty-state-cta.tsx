import { Button } from '@automattic/components';
import { MEDIA_QUERIES } from '../utils';

interface EmptyStateCTAProps {
	heading: string;
	description: string;
	cta: string;
	target: string;
}

export const EmptyStateCTA = ( { heading, description, cta, target }: EmptyStateCTAProps ) => {
	return (
		<div
			css={ {
				display: 'flex',
				justifyContent: 'space-between',
				alignItems: 'center',
				[ MEDIA_QUERIES.small ]: {
					alignItems: 'flex-start',
					flexDirection: 'column',
					gap: '24px',
				},
			} }
		>
			<div
				css={ {
					display: 'flex',
					flexDirection: 'column',
					gap: '8px',
					alignItems: 'flex-start',
					marginRight: '32px',
				} }
			>
				<h2
					css={ {
						fontSize: '20px',
						lineHeight: '24px',
						color: '#101517',
					} }
				>
					{ heading }
				</h2>
				<span
					css={ {
						fontSize: '14px',
						lineHeight: '20px',
						color: '#646970',
					} }
				>
					{ description }
				</span>
			</div>
			<Button href={ target } primary>
				{ cta }
			</Button>
		</div>
	);
};
