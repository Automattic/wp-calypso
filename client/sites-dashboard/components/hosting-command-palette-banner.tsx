import { Gridicon } from '@automattic/components';
import { keyframes } from '@emotion/react';
import styled from '@emotion/styled';
import { useMediaQuery } from '@wordpress/compose';
import {
	globe as domainsIcon,
	plus as plusIcon,
	settings as settingsIcon,
	tool as toolIcon,
	wordpress as wordpressIcon,
} from '@wordpress/icons';
import { useI18n } from '@wordpress/react-i18n';
import classNames from 'classnames';
import { useTranslate } from 'i18n-calypso';
import { useEffect, useState } from 'react';
import DismissibleCard from 'calypso/blocks/dismissible-card';

const HostingCommandPaletteBannerRoot = styled.div( {
	'&:not(:empty)': {
		marginBottom: 25,
	},
	'.hosting-command-palette-banner': {
		background: 'linear-gradient(270deg, #E9EFF5 12.03%, rgba(233, 239, 245, 0) 40.39%)',
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'left',
		borderLeftWidth: 3,
		borderLeftColor: 'var(--color-accent)',
		borderLeftStyle: 'solid',
		padding: '10px 56px 10px 12px',
	},
	'.dismissible-card__close-button': {
		top: 'calc(50% - 12px)',
		right: 20,
	},
} );

const fadeIn = keyframes`
  0% {
    transform: translateX(15%);
    opacity: 0;
  }
  50% {
	transform: translateX(-1%);
	opacity: 1;
  }
  100% {
	opacity: 1,
    transform: translateX(0);
  }
`;

const fadeOut = keyframes`
  0% {
    transform: translateX(0%);
	opacity: 1;
  }
  50% {
    opacity: 0;
  }
  100% {
    opacity: 0;
	transform: translateX(-50%);
  }
`;

const CommandBoxWrapper = styled.div( {
	marginLeft: 'auto',
	display: 'grid',
	gridTemplateColumns: '1fr',
	flexGrow: 1,
	userSelect: 'none',
	'@media ( max-width: 810px )': {
		display: 'none',
	},
} );

const CommandBox = styled.div( {
	gridRowStart: 1,
	gridColumnStart: 1,
	borderRadius: 2,
	display: 'flex',
	justifyContent: 'flex-start',
	alignItems: 'center',
	marginLeft: 'auto',
	minWidth: 253,
	fontSize: 13,
	lineHeight: '20px',
	letterSpacing: -0.15,
	backgroundColor: 'var(--color-surface)',
	boxShadow: '1px 1px 1px 0px #0000001F',
	paddingRight: 10,
	whiteSpace: 'nowrap',
	'&.command-box__fadeIn': {
		animation: `${ fadeIn } 1s ease-in-out`,
		zIndex: 10,
	},
	'&.command-box__fadeOut': {
		animation: `${ fadeOut } 1s ease-in-out`,
		zIndex: 5,
		opacity: 0,
	},
	svg: {
		width: 24,
		height: 24,
		margin: 8,
	},
} );

const BannerTitle = styled.div( {
	fontSize: 14,
	fontWeight: 600,
} );

const BannerDescription = styled.div( {
	color: 'var(--Gray-Gray-70, #3C434A)',
	fontSize: 14,
	paddingRight: 20,
} );

interface HostingCommandPaletteBannerProps {
	className?: string;
}

interface ShortcutIconProps {
	size: 'big' | 'small';
}

const StyledShortcut = styled.span< ShortcutIconProps >( ( props ) =>
	props.size === 'big'
		? {
				fontWeight: 600,
				fontSize: 20,
				border: '1px solid',
				borderRadius: 2,
				padding: '0px 8px',
				marginRight: 12,
		  }
		: {
				fontWeight: 600,
				fontSize: 14,
				backgroundColor: '#EEE',
				color: '#3C434A',
				padding: '0 4px',
				borderRadius: 2,
		  }
);

const ShortcutIcon = ( { size }: ShortcutIconProps ) => {
	const isMac = navigator.userAgent.indexOf( 'Mac' ) > -1;
	const shortcut = isMac ? '⌘K' : 'Ctrl+K';
	return <StyledShortcut size={ size }>{ shortcut }</StyledShortcut>;
};

const AnimatedCommand = () => {
	const { __ } = useI18n();
	const prefersReducedMotion = window.matchMedia( '(prefers-reduced-motion: reduce)' ).matches;
	const commands = [
		{ icon: domainsIcon, label: __( 'Register new domain' ) },
		{ icon: <Gridicon icon="reader" />, label: __( 'Open reader' ) },
		{ icon: plusIcon, label: __( 'Add new post' ) },
		{ icon: wordpressIcon, label: __( 'View my sites' ) },
		{ icon: settingsIcon, label: __( 'Open hosting configuration' ) },
		{ icon: toolIcon, label: __( 'Change PHP version' ) },
		{ icon: toolIcon, label: __( 'Manage staging sites' ) },
	];

	const [ currentCommandIndex, setCurrentCommandIndex ] = useState( 0 );

	useEffect( () => {
		if ( prefersReducedMotion ) {
			return;
		}

		const interval = setInterval( () => {
			setCurrentCommandIndex( ( prevIndex ) => ( prevIndex + 1 ) % commands.length );
		}, 3000 );

		return () => clearInterval( interval );
	}, [ prefersReducedMotion, commands.length ] );

	if ( prefersReducedMotion ) {
		return null;
	}
	return (
		<CommandBoxWrapper>
			{ commands.map( ( commandName, index ) => {
				const prevIndex = ( currentCommandIndex - 1 + commands.length ) % commands.length;
				if ( currentCommandIndex !== index && prevIndex !== index ) {
					return null;
				}

				return (
					<CommandBox
						className={ classNames( {
							'command-box__fadeIn': currentCommandIndex === index,
							'command-box__fadeOut': prevIndex === index,
						} ) }
						key={ index }
					>
						{ commandName?.icon }
						{ commandName?.label }
					</CommandBox>
				);
			} ) }
		</CommandBoxWrapper>
	);
};

export function HostingCommandPaletteBanner( { className }: HostingCommandPaletteBannerProps ) {
	const translate = useTranslate();
	const isSmallScreen = useMediaQuery( '(max-width: 600px)' );
	if ( isSmallScreen ) {
		return null;
	}

	return (
		<HostingCommandPaletteBannerRoot>
			<DismissibleCard
				preferenceName="hosting-command-palette-banner-display"
				className={ `hosting-command-palette-banner is-dismissible ${ className }` }
			>
				<ShortcutIcon size="big" />
				<div>
					<BannerTitle>{ translate( 'The Command Palette is here' ) }</BannerTitle>
					<BannerDescription>
						{ translate(
							// Translators: {{shortcut/}} is "⌘K" or "Ctrl+K" depending on the user's OS.
							'Access features and trigger commands quickly. Press {{shortcut/}} to launch the Command Palette.',
							{
								components: { shortcut: <ShortcutIcon size="small" /> },
							}
						) }
					</BannerDescription>
				</div>
				<AnimatedCommand />
			</DismissibleCard>
		</HostingCommandPaletteBannerRoot>
	);
}
