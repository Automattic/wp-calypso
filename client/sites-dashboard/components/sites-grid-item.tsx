import { Button } from '@automattic/components';
import { useSiteLaunchStatusLabel, getSiteLaunchStatus } from '@automattic/sites';
import { css } from '@emotion/css';
import styled from '@emotion/styled';
import { useI18n } from '@wordpress/react-i18n';
import { memo } from 'react';
import { useInView } from 'react-intersection-observer';
import SitesMigrationTrialBadge from 'calypso/sites-dashboard/components/sites-migration-trial-badge';
import { useSelector } from 'calypso/state';
import { isTrialSite } from 'calypso/state/sites/plans/selectors';
import {
	displaySiteUrl,
	getDashboardUrl,
	isStagingSite,
	siteDefaultInterface,
	getSiteWpAdminUrl,
} from '../utils';
import { SitesEllipsisMenu } from './sites-ellipsis-menu';
import { SitesGridActionRenew } from './sites-grid-action-renew';
import { SitesGridTile } from './sites-grid-tile';
import SitesLaunchStatusBadge from './sites-launch-status-badge';
import SitesP2Badge from './sites-p2-badge';
import { SiteItemThumbnail } from './sites-site-item-thumbnail';
import { SiteLaunchNag } from './sites-site-launch-nag';
import { SiteName } from './sites-site-name';
import { SiteUrl, Truncated } from './sites-site-url';
import SitesStagingBadge from './sites-staging-badge';
import TransferNoticeWrapper from './sites-transfer-notice-wrapper';
import { ThumbnailLink } from './thumbnail-link';
import { WithAtomicTransfer } from './with-atomic-transfer';
import type { SiteExcerptData } from '@automattic/sites';

const SIZES_ATTR = [
	'(min-width: 1345px) calc((1280px - 64px) / 3)',
	'(min-width: 960px) calc((100vw - 128px) / 3)',
	'(min-width: 780px) calc((100vw - 96px) / 2)',
	'(min-width: 660px) calc((100vw - 64px) / 2)',
	'calc(100vw - 32px)',
].join( ', ' );

const ASPECT_RATIO = 16 / 11;

const THUMBNAIL_DIMENSION = {
	width: 401,
	height: 401 / ASPECT_RATIO,
};

const badges = css( {
	display: 'flex',
	gap: '8px',
	alignItems: 'center',
	marginInlineStart: 'auto',
} );

const selectAction = css( {
	display: 'flex',
	gap: '8px',
	alignItems: 'center',
	marginInlineStart: 'auto',
	button: {
		whiteSpace: 'nowrap',
	},
} );

export const siteThumbnail = css( {
	aspectRatio: '16 / 11',
	width: '100%',
	height: 'auto',
	boxSizing: 'border-box',
} );

const SitesGridItemSecondary = styled.div( {
	display: 'flex',
	gap: '20px',
	fontSize: '14px',
	justifyContent: 'space-between',
} );

const EllipsisMenuContainer = styled.div( {
	width: '24px',
} );

const ellipsis = css( {
	'.button.ellipsis-menu__toggle': {
		padding: 0,
	},

	'.gridicon.ellipsis-menu__toggle-icon': {
		width: '24px',
		height: '16px',
		insetBlockStart: '4px',
	},
} );

interface SitesGridItemProps {
	site: SiteExcerptData;
	showLaunchNag?: boolean;
	showBadgeSection?: boolean;
	showThumbnailLink?: boolean;
	showSiteRenewLink?: boolean;
	onSiteSelectBtnClick?: ( site: SiteExcerptData ) => void;
}

export const SitesGridItem = memo( ( props: SitesGridItemProps ) => {
	const { __ } = useI18n();
	const {
		site,
		showLaunchNag = true,
		showBadgeSection = true,
		showThumbnailLink = true,
		showSiteRenewLink = true,
		onSiteSelectBtnClick,
	} = props;

	const isP2Site = site.options?.is_wpforteams_site;
	const isWpcomStagingSite = isStagingSite( site );
	const translatedStatus = useSiteLaunchStatusLabel( site );
	const isTrialSitePlan = useSelector( ( state ) => isTrialSite( state, site.ID ) );
	const wpAdminUrl = getSiteWpAdminUrl( site );
	const { ref, inView } = useInView( { triggerOnce: true } );

	const ThumbnailWrapper = showThumbnailLink ? ThumbnailLink : 'div';

	const siteDashboardUrlProps = showThumbnailLink
		? {
				href:
					siteDefaultInterface( site ) === 'wp-admin'
						? wpAdminUrl || getDashboardUrl( site.slug )
						: getDashboardUrl( site.slug ),
				title: __( 'Visit Dashboard' ),
		  }
		: {};

	let siteUrl = site.URL;
	if ( site.options?.is_redirect && site.options?.unmapped_url ) {
		siteUrl = site.options?.unmapped_url;
	}
	return (
		<SitesGridTile
			ref={ ref }
			leading={
				<>
					<ThumbnailWrapper { ...siteDashboardUrlProps }>
						<SiteItemThumbnail
							displayMode="tile"
							className={ siteThumbnail }
							showPlaceholder={ ! inView }
							site={ site }
							width={ THUMBNAIL_DIMENSION.width }
							height={ THUMBNAIL_DIMENSION.height }
							sizesAttr={ SIZES_ATTR }
						/>
					</ThumbnailWrapper>
					{ showSiteRenewLink && site.plan?.expired && (
						<SitesGridActionRenew site={ site } isUpgradeable={ isTrialSitePlan } />
					) }
				</>
			}
			primary={
				<>
					<SiteName fontSize={ 16 } { ...siteDashboardUrlProps }>
						{ site.title }
					</SiteName>

					{ showBadgeSection && (
						<div className={ badges }>
							{ isP2Site && <SitesP2Badge>P2</SitesP2Badge> }
							{ isWpcomStagingSite && <SitesStagingBadge>{ __( 'Staging' ) }</SitesStagingBadge> }
							{ isTrialSitePlan && (
								<SitesMigrationTrialBadge>{ __( 'Trial' ) }</SitesMigrationTrialBadge>
							) }
							{ getSiteLaunchStatus( site ) !== 'public' && ! isTrialSitePlan && (
								<SitesLaunchStatusBadge>{ translatedStatus }</SitesLaunchStatusBadge>
							) }
							<EllipsisMenuContainer>
								{ inView && <SitesEllipsisMenu className={ ellipsis } site={ site } /> }
							</EllipsisMenuContainer>
						</div>
					) }
					{ onSiteSelectBtnClick && (
						<div className={ selectAction }>
							<Button
								compact={ true }
								primary={ true }
								onClick={ () => onSiteSelectBtnClick( site ) }
							>
								{ __( 'Select this site' ) }
							</Button>
						</div>
					) }
				</>
			}
			secondary={
				<SitesGridItemSecondary>
					<SiteUrl href={ siteUrl } title={ siteUrl }>
						<Truncated>{ displaySiteUrl( siteUrl ) }</Truncated>
					</SiteUrl>
					<WithAtomicTransfer site={ site }>
						{ ( result ) => {
							if ( result.wasTransferring ) {
								return <TransferNoticeWrapper { ...result } />;
							} else if ( showLaunchNag && 'unlaunched' === site.launch_status ) {
								return <SiteLaunchNag site={ site } />;
							}
							return <></>;
						} }
					</WithAtomicTransfer>
				</SitesGridItemSecondary>
			}
		/>
	);
} );
