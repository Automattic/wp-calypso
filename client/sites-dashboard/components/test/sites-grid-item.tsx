/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { SitesGridItem } from 'calypso/sites-dashboard/components/sites-grid-item';
import { useCheckSiteTransferStatus } from '../../hooks/use-check-site-transfer-status';
import type { SiteExcerptData } from '@automattic/sites';

function makeTestSite( { title = 'test', is_coming_soon = false, lang = 'en' } = {} ) {
	return {
		ID: 1,
		title,
		slug: 'test_slug',
		URL: '',
		launch_status: 'launched',
		options: {},
		jetpack: false,
		is_coming_soon,
		lang,
	};
}

jest.mock( 'calypso/sites-dashboard/hooks/use-check-site-transfer-status.tsx', () => ( {
	__esModule: true,
	useCheckSiteTransferStatus: jest.fn(),
} ) );

jest.mock( 'react-redux', () => ( {
	...jest.requireActual( 'react-redux' ),
	useSelector: jest.fn(),
} ) );

describe( '<SitesGridItem>', () => {
	beforeEach( () => {
		( useCheckSiteTransferStatus as jest.Mock ).mockReturnValue( {
			isTransferInProgress: false,
		} );
	} );
	test( 'Default render', () => {
		const tree = renderer
			.create(
				<SitesGridItem site={ makeTestSite( { title: 'The example site' } ) as SiteExcerptData } />
			)
			.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	test( 'Custom render', () => {
		const tree = renderer
			.create(
				<SitesGridItem
					site={ makeTestSite( { title: 'The example site' } ) as SiteExcerptData }
					showLaunchNag={ false }
					showBadgeSection={ false }
					showThumbnailLink={ true }
					showSiteRenewLink={ false }
				/>
			)
			.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	test( 'Custom render 2', () => {
		const tree = renderer
			.create(
				<SitesGridItem
					site={ makeTestSite( { title: 'The example site' } ) as SiteExcerptData }
					showLaunchNag={ false }
					showBadgeSection={ false }
					showThumbnailLink={ false }
					showSiteRenewLink={ false }
				/>
			)
			.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
