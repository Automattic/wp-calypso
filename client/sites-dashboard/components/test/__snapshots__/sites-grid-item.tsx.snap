// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SitesGridItem> Custom render 1`] = `
<div
  className="css-lvc9r-container"
>
  <a
    className="css-1ctre4x-ThumbnailLink e19s2eol0"
    href="/home/test_slug"
    title="Visit Dashboard"
  >
    <div
      className="site-thumbnail site-thumbnail-loading css-ut2tya-DEFAULT_CLASSNAME css-1mwzofc-siteThumbnail css-1uizeh6-disallowSelection"
      style={
        Object {
          "backgroundColor": undefined,
          "color": undefined,
        }
      }
    >
      <div
        className="site-thumbnail-icon"
      >
        <div
          aria-label="Site Icon"
          className="css-k8j3dg-NoIcon e1y9o06l0"
          role="img"
        >
          T
        </div>
      </div>
    </div>
  </a>
  <div>
    <div
      className="css-1xtz7v6-primaryContainer"
    >
      <a
        className="css-6u0p2y-SiteName esmj6c50"
        fontSize={16}
        href="/home/test_slug"
        title="Visit Dashboard"
      >
        The example site
      </a>
    </div>
    <div
      className="css-12bwudn-SitesGridItemSecondary eke2d2q1"
    >
      <a
        className="components-external-link css-d9p5eo-SiteUrl ejzgyku1"
        href=""
        onClick={[Function]}
        rel="external noreferrer noopener"
        target="_blank"
        title=""
      >
        <span
          className="css-1i2o39r-Truncated ejzgyku0"
        />
        <span
          className="components-visually-hidden css-1mm2cvy-View e19lxcc00"
          data-wp-c16t={true}
          data-wp-component="VisuallyHidden"
          style={
            Object {
              "WebkitClipPath": "inset( 50% )",
              "border": 0,
              "clip": "rect(1px, 1px, 1px, 1px)",
              "clipPath": "inset( 50% )",
              "height": "1px",
              "margin": "-1px",
              "overflow": "hidden",
              "padding": 0,
              "position": "absolute",
              "width": "1px",
              "wordWrap": "normal",
            }
          }
        >
          (opens in a new tab)
        </span>
        <svg
          aria-hidden={true}
          className="components-external-link__icon css-16iaek2-StyledIcon esh4a730"
          focusable={false}
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.2 17c0 .7-.6 1.2-1.2 1.2H7c-.7 0-1.2-.6-1.2-1.2V7c0-.7.6-1.2 1.2-1.2h3.2V4.2H7C5.5 4.2 4.2 5.5 4.2 7v10c0 1.5 1.2 2.8 2.8 2.8h10c1.5 0 2.8-1.2 2.8-2.8v-3.6h-1.5V17zM14.9 3v1.5h3.7l-6.4 6.4 1.1 1.1 6.4-6.4v3.7h1.5V3h-6.3z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`<SitesGridItem> Custom render 2 1`] = `
<div
  className="css-lvc9r-container"
>
  <div>
    <div
      className="site-thumbnail site-thumbnail-loading css-ut2tya-DEFAULT_CLASSNAME css-1mwzofc-siteThumbnail css-1uizeh6-disallowSelection"
      style={
        Object {
          "backgroundColor": undefined,
          "color": undefined,
        }
      }
    >
      <div
        className="site-thumbnail-icon"
      >
        <div
          aria-label="Site Icon"
          className="css-k8j3dg-NoIcon e1y9o06l0"
          role="img"
        >
          T
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="css-1xtz7v6-primaryContainer"
    >
      <a
        className="css-6u0p2y-SiteName esmj6c50"
        fontSize={16}
      >
        The example site
      </a>
    </div>
    <div
      className="css-12bwudn-SitesGridItemSecondary eke2d2q1"
    >
      <a
        className="components-external-link css-d9p5eo-SiteUrl ejzgyku1"
        href=""
        onClick={[Function]}
        rel="external noreferrer noopener"
        target="_blank"
        title=""
      >
        <span
          className="css-1i2o39r-Truncated ejzgyku0"
        />
        <span
          className="components-visually-hidden css-1mm2cvy-View e19lxcc00"
          data-wp-c16t={true}
          data-wp-component="VisuallyHidden"
          style={
            Object {
              "WebkitClipPath": "inset( 50% )",
              "border": 0,
              "clip": "rect(1px, 1px, 1px, 1px)",
              "clipPath": "inset( 50% )",
              "height": "1px",
              "margin": "-1px",
              "overflow": "hidden",
              "padding": 0,
              "position": "absolute",
              "width": "1px",
              "wordWrap": "normal",
            }
          }
        >
          (opens in a new tab)
        </span>
        <svg
          aria-hidden={true}
          className="components-external-link__icon css-16iaek2-StyledIcon esh4a730"
          focusable={false}
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.2 17c0 .7-.6 1.2-1.2 1.2H7c-.7 0-1.2-.6-1.2-1.2V7c0-.7.6-1.2 1.2-1.2h3.2V4.2H7C5.5 4.2 4.2 5.5 4.2 7v10c0 1.5 1.2 2.8 2.8 2.8h10c1.5 0 2.8-1.2 2.8-2.8v-3.6h-1.5V17zM14.9 3v1.5h3.7l-6.4 6.4 1.1 1.1 6.4-6.4v3.7h1.5V3h-6.3z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`<SitesGridItem> Default render 1`] = `
<div
  className="css-lvc9r-container"
>
  <a
    className="css-1ctre4x-ThumbnailLink e19s2eol0"
    href="/home/test_slug"
    title="Visit Dashboard"
  >
    <div
      className="site-thumbnail site-thumbnail-loading css-ut2tya-DEFAULT_CLASSNAME css-1mwzofc-siteThumbnail css-1uizeh6-disallowSelection"
      style={
        Object {
          "backgroundColor": undefined,
          "color": undefined,
        }
      }
    >
      <div
        className="site-thumbnail-icon"
      >
        <div
          aria-label="Site Icon"
          className="css-k8j3dg-NoIcon e1y9o06l0"
          role="img"
        >
          T
        </div>
      </div>
    </div>
  </a>
  <div>
    <div
      className="css-1xtz7v6-primaryContainer"
    >
      <a
        className="css-6u0p2y-SiteName esmj6c50"
        fontSize={16}
        href="/home/test_slug"
        title="Visit Dashboard"
      >
        The example site
      </a>
      <div
        className="css-lnphaf-badges"
      >
        <div
          className="css-1j6mrwp-EllipsisMenuContainer eke2d2q0"
        />
      </div>
    </div>
    <div
      className="css-12bwudn-SitesGridItemSecondary eke2d2q1"
    >
      <a
        className="components-external-link css-d9p5eo-SiteUrl ejzgyku1"
        href=""
        onClick={[Function]}
        rel="external noreferrer noopener"
        target="_blank"
        title=""
      >
        <span
          className="css-1i2o39r-Truncated ejzgyku0"
        />
        <span
          className="components-visually-hidden css-1mm2cvy-View e19lxcc00"
          data-wp-c16t={true}
          data-wp-component="VisuallyHidden"
          style={
            Object {
              "WebkitClipPath": "inset( 50% )",
              "border": 0,
              "clip": "rect(1px, 1px, 1px, 1px)",
              "clipPath": "inset( 50% )",
              "height": "1px",
              "margin": "-1px",
              "overflow": "hidden",
              "padding": 0,
              "position": "absolute",
              "width": "1px",
              "wordWrap": "normal",
            }
          }
        >
          (opens in a new tab)
        </span>
        <svg
          aria-hidden={true}
          className="components-external-link__icon css-16iaek2-StyledIcon esh4a730"
          focusable={false}
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.2 17c0 .7-.6 1.2-1.2 1.2H7c-.7 0-1.2-.6-1.2-1.2V7c0-.7.6-1.2 1.2-1.2h3.2V4.2H7C5.5 4.2 4.2 5.5 4.2 7v10c0 1.5 1.2 2.8 2.8 2.8h10c1.5 0 2.8-1.2 2.8-2.8v-3.6h-1.5V17zM14.9 3v1.5h3.7l-6.4 6.4 1.1 1.1 6.4-6.4v3.7h1.5V3h-6.3z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
